.github/workflows/build.yml

```diff
@@ -0,0 +1,26 @@
+name: build
+
+on: [push]
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macOS-latest]
+        rust: [stable, nightly]
+
+    steps:
+    - uses: hecrj/setup-rust-action@v1
+      with:
+        rust-version: ${{ matrix.rust }}
+        components: rustfmt, clippy
+    - uses: actions/checkout@master
+    - name: Check Formatting
+      run: cargo fmt --all -- --check
```

Comment:
I would rather add fmt check as a separate job, no need to run it for each os
something like this: https://github.com/iqlusioninc/crates/blob/develop/.github/workflows/rust.yml